R/matchDim.R

```diff
@@ -35,3 +34,57 @@ matchDim <- function(x, ref, fill = 0) {
 
   return(r)
 }
+
+#' match spatial dimension of a magclass object
+#'
+#' A helper that restricts and expands a magclass object x to the size of a
+#' magclass object ref in the spatial dimension.
+#'
+#' @param x a magclass object to be modified
+#' @param ref a magclass object used as a reference for the modification
+#' @param fill value to be set in new dimensions
+#' @importFrom magclass ndim
+#'
+#' @export
+matchRegions <- function(x, ref, fill = 0) {
+  r <- new.magpie(
+    cells_and_regions = union(getRegions(x), getRegions(ref)),
+    years = getYears(x),
+    names = getNames(x),
+    fill = fill,
+    sets = names(dimnames(x))
+  )
+
+  # copy over values from x
+  r[getRegions(x), , ] <- x
+
+  # restrict object to dimensions of ref
+  r <- r[getRegions(ref), , ]
+}
+
+#' match temporal dimension of a magclass object
+#'
+#' A helper that restricts and expands a magclass object x to the size of a
+#' magclass object ref in the temporal dimension.
+#'
+#' @param x a magclass object to be modified
+#' @param ref a magclass object used as a reference for the modification
+#' @param fill value to be set in new dimensions
+#' @importFrom magclass ndim
+#'
+#' @export
+matchYears <- function(x, ref, fill = 0) {
```

Comment:
If one needs `NA`, one can pass it as a parameter.  But the default operation for this will be to extend some magpie object to be summed up to some other magpie object, and 0 is the neutral element for addition.